src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -43,38 +43,46 @@ public void apply(Object target) {
         }
     }
 
-    private void applySettingsPlugin(Settings settings) {
-        settings.getPluginManager().withPlugin("com.gradle.enterprise", __ -> {
-            CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
-
-            Object extension = settings.getExtensions().getByName("gradleEnterprise");
-            GradleEnterpriseExtension gradleEnterprise = ProxyFactory.createProxy(extension, GradleEnterpriseExtension.class);
-            customGradleEnterpriseConfig.configureGradleEnterprise(gradleEnterprise);
-
-            BuildScanExtension buildScan = gradleEnterprise.getBuildScan();
-            customGradleEnterpriseConfig.configureBuildScanPublishing(buildScan);
-            CustomBuildScanEnhancements buildScanEnhancements = new CustomBuildScanEnhancements(buildScan, providers, settings.getGradle());
-            buildScanEnhancements.apply();
-
-            BuildCacheConfiguration buildCache = settings.getBuildCache();
-            customGradleEnterpriseConfig.configureBuildCache(buildCache);
+    public static void apply(GradleEnterpriseExtension gradleEnterprise, ProviderFactory providers, Settings settings) {
+        applySettingsPlugin(
+                gradleEnterprise,
+                providers,
+                settings
+        );
+    }
 
-            // configuration changes applied in this block will override earlier configuration settings,
-            // including those set in the settings.gradle(.kts)
-            Action<Settings> settingsAction = ___ -> {
-                Overrides overrides = new Overrides(providers);
-                overrides.configureGradleEnterprise(gradleEnterprise);
-                overrides.configureBuildCache(buildCache);
-            };
+    private static void applySettingsPlugin(GradleEnterpriseExtension gradleEnterprise, ProviderFactory providers, Settings settings) {
```

Comment:
You could change this method to take `Object gradleEnterprise` and do the proxying internally. That would simplify our use-site, so we won't have to duplicate the proxying.

Reply:
I've just seen that we don't actually need proxying with our usage currently. I still recommend making this change though in case we want to change things to only load the CCUD plugin once.

Reply:
OK will do
The interesting bit about loading it twice is that we could dogfood dev version of it while keeping a stable version for our stable build scans

Reply:
done